### PR TITLE
Recurrence appointments

### PR DESCRIPTION
**Pull request by vinay-ThoughtWorks**

### CODE REVIEW

File: ui/app/appointments/controllers/manage/appointmentsCreateController.js [hunk 1]
```diff
@@ -54,9 +54,15 @@ angular.module('bahmni.appointments')
             var setRecurringToTrueForRecurringAppointments = function () {
                 if (!_.isEmpty($scope.appointment.recurringPattern)) {
                     $scope.appointment.setRecurring = true;
+                    $scope.updateRecurrenceInstance('');
```
Comment (bsneha90) [inline ui/app/appointments/controllers/manage/appointmentsCreateController.js]: Does this mean the selected recurrence instance?

File: ui/app/appointments/controllers/manage/appointmentsCreateController.js [hunk 2]
```diff
@@ -69,6 +75,7 @@ angular.module('bahmni.appointments')
                 $scope.appointment = Bahmni.Appointments.AppointmentViewModel.create(appointmentContext.appointment || {appointmentKind: 'Scheduled'}, appointmentCreateConfig);
                 if ($scope.isEditMode()) {
                     setRecurringToTrueForRecurringAppointments();
+                    $scope.previousAppointmentStartDate = $scope.appointment.date;
```
Comment (bsneha90) [inline ui/app/appointments/controllers/manage/appointmentsCreateController.js]: Does this need to be stored in scope or can it just be a locally scoped variable?

File: ui/app/appointments/controllers/manage/appointmentsCreateController.js (hunk 1 above)
Comment (vinay-ThoughtWorks) [reply ui/app/appointments/controllers/manage/appointmentsCreateController.js]: Yes.

File: ui/app/appointments/controllers/manage/appointmentsCreateController.js (hunk 2 above)
Comment (vinay-ThoughtWorks) [reply ui/app/appointments/controllers/manage/appointmentsCreateController.js]: better to be in scope. If we separate the instances(thisAppointment, allAppointments) as separate component, we would need this information in that component to disable/enable.

File: ui/app/appointments/controllers/manage/appointmentsCreateController.js (hunk 1 above)
Comment (bsneha90) [reply ui/app/appointments/controllers/manage/appointmentsCreateController.js]: Can we rename it to selectedAppointment?

File: ui/app/appointments/controllers/manage/appointmentsCreateController.js (hunk 2 above)
Comment (bsneha90) [reply ui/app/appointments/controllers/manage/appointmentsCreateController.js]: ok, can we introduce it then. Ideally scope is used for binding between ui and controller, not within the controller methods.

**Pull request merged**